phone-test/src/gql/client/index.ts

```diff
@@ -68,7 +30,44 @@ const authLink = setContext((_, { headers }) => {
   };
 });
 
+const errorLink: ApolloLink = onError(({ graphQLErrors, operation, forward }) => {
+  let promise = Promise.resolve();
+  let isRefreshingToken: boolean = false;
+
+  if (graphQLErrors?.some(err => err.message === 'Unauthorized')) {
+    const isWebsocket = isWebsocketSubscription(operation);
+    if (!isRefreshingToken) {
+      isRefreshingToken = true;
+      promise = getRefreshedToken(client)
+        .then(() => {
+          isRefreshingToken = false;
+          if (isWebsocket) {
+            wsClient.close(true);
+          }
+        })
+        .catch(() => {
+          localStorage.clear();
+          window.location.reload();
+        });
+    } else {
+      return forward(operation);
+    }
+
+    return fromPromise(promise).flatMap(() => {
+      return forward(operation);
+    });
+  }
+});
+
+const operationLinks: ApolloLink = split(
+  operation => {
+    return isWebsocketSubscription(operation);
+  },
+  wsLink,
+  httpLink
+);
```

Comment:
We want to use the relevant link, depending on whether or not the operation is targeting the normal or the websocket endpoint